### ISSUE TITLE
[Spike] Will Glitchtip work with MykoMaps?

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/mykomap#250]
## Description

In investigations [here](https://github.com/DigitalCommons/land-explorer-front-end/issues/202) and [here](https://docs.google.com/document/d/1vEm8orZ6D3DPUvGYMJtjcWjOg0BLc1VUMS67QYLCD6I/edit) we've identified the need from persistent error logging to help us track down and minimise bug occurances in LX. The tool that seems the most fitting, balancing functionality, OSS desires and implementation weight is [GlitchTip](https://glitchtip.com/). In the spirit of unifying our pipe across our projects we'd like to make sure this will work with MM as well as LX.

## Acceptance Criteria

An estimate for linking [Glitchtip](https://glitchtip.com/) to MM, so that errors triggered in a production MM are logged in GlitchTip. This might be per map or it might work for any MM hosted on the same server, I don't know how it will work in practice. For context the LX estimate is 1-2 days.

**Comment by wu-lee** [comment]
A quick look GitchTip suggests:
- Not available on Cloudron (although there is a request [here][1])
- Self-hosting ourselves might be possible but we don't currently have any support for docker apps.
- Therefore suspect using the cloud service is probably most expedient.
- It [says here][2] that "GlitchTip is an open-source alternative to Sentry that works seamlessly with the Sentry browser SDK."
  -  sentry.io is a general tracing tool/service, with support for a wide range of platforms, but not open source
- For JS client-side apps like MM, you add @browser/sentry to the npm deps of the app, and initialise it to report to GlitchTip early in the app's lifecycle.  Errors are then reported to the server and can be analysed.
- It can also be used for uptime reporting (not relevant for MM as a client-side app) and error alerts (maybe useful for spotting when MM loses it unexpectedly in general use)

In summary, yes I think it can be used with MM but to make sure we should try it out.


[1]: https://forum.cloudron.io/topic/5246/error-tracking-glitchtip
[2]: https://glitchtip.com/sdkdocs/javascript
